src/typeform-types.ts

```diff
@@ -0,0 +1,1233 @@
+       */
+      href?: string
+    } | string
+    /**
+     * Workspace that contains the form.
+     */
+    workspace?: {
+      /**
+       * URL of the workspace to use for the typeform.
+       * If you don't specify a URL for the workspace, Typeform saves the form in the default workspace.
+       */
+      href?: string
+    }
+    /**
+     * Only available when retrieving a form.
+     */
+    _links?: {
+      display?: string
+    }
+    /**
+     * Object that specifies form settings and metadata, including the language to use for the form,
+     * whether the form is publicly available, the basis for the progress bar, and search engine indexing settings.
+     */
+    settings?: Settings
+    /**
+     * Object that keeps track of total score or price, if you use them in the form.
+     * Not available when retrieving a form.
+     */
+    variables?: {
+      /**
+       * Recall Information for keeping score as users answer each question (for example, for quizzes).
+       */
+      score?: 0
+      /**
+       * Recall Information for tracking the total price of all items users select
+       * (for example, for shopping carts, donation campaigns, and payment collections).
+       */
+      price?: number
+    }
+  }
+  /**
+   * HTTP Client for API requests.
+   */
+  export interface HTTPClient {
+    request: (args: Request) => Promise<any>
+  }
+  /**
+   * Typeform Image object.
+   */
+  export interface Image {
+    /**
+     * Unique ID for the image.
+     */
+    id?: string
+    /**
+     * URL for the image.
+     */
+    src?: string
+    /**
+     * File name for the image (specified when image is created).
+     */
+    file_name?: string
+    /**
+     * Width of the image in pixels.
+     */
+    width?: number
+    /**
+     * Height of the image in pixels.
+     */
+    height?: number
+    /**
+     * The MIME type of the image.
+     */
+    media_type?: 'image/gif' | 'image/jpeg' | 'image/png'
+    /**
+     * True if image has alpha channel (some degree of transparency). Otherwise, false.
+     */
+    has_alpha?: boolean
+    /**
+     * Average color of the image in hexadecimal format.
+     */
+    avg_color?: string
+  }
+  /**
+   * Language that Typeform can be in.
+   */
+  export type Language = 'en' | 'es' | 'ca' | 'fr' | 'de' | 'ru' | 'it' | 'da' | 'pt' | 'ch' | 'zh' | 'nl' | 'no' | 'uk' | 'ja' | 'ko'
+    | 'hr' | 'fi' | 'sv' | 'pl' | 'el' | 'hu' | 'tr' | 'cs' | 'et' | 'di'
+  /**
+   * Logic object of a form.
+   */
+  export interface Logic {
+    /**
+     * Specifies whether the Logic Jump is based on a question field or Hidden Field.
+     */
+    type?: 'field' | 'hidden'
+    /**
+     * Reference to the field that triggers the the Logic Jump.
+     */
+    ref?: string
+    /**
+     * Array of objects that define the Logic Jump's behavior.
+     */
+    actions?: Action[]
+  }
+  /**
+   * Messages that forms can use.
+   */
+  export interface Messages {
```

Comment:
This is amazing, with the descriptions 🎉 

However, is there maybe a way we generate these messages types programatically, if they're prone to change (and I assume it's the API who sets them)?

Reply:
I think the intention is to automate some of this down the road, but time/bandwidth/resources are lacking on my end. I know @picsoung wants to do this. I don't think that the definitions of file structures are likely to change as quickly as the API endpoints are (like new endpoints and new functionality)